Branch-SDK/src/main/java/io/branch/referral/SystemObserver.java

```diff
@@ -600,18 +600,18 @@ public CoroutineContext getContext() {
 
                 @Override
                 public void resumeWith(@NonNull Object o) {
-                    if (o != null) {
-                        InstallReferrerResult latestReferrer = (InstallReferrerResult) o;
-                        AppStoreReferrer.processReferrerInfo(context_, latestReferrer.getLatestRawReferrer(), latestReferrer.getLatestClickTimestamp(), latestReferrer.getLatestInstallTimestamp(), latestReferrer.getAppStore(), latestReferrer.isClickThrough());
+                    BranchLogger.v("fetchInstallReferrer resumeWith got result: " + o);
+                    InstallReferrerResult latestReferrer = (InstallReferrerResult) o;
+                    AppStoreReferrer.processReferrerInfo(context_, latestReferrer.getLatestRawReferrer(), latestReferrer.getLatestClickTimestamp(), latestReferrer.getLatestInstallTimestamp(), latestReferrer.getAppStore(), latestReferrer.isClickThrough());
+
+                    if (callback != null) {
+                        callback.onInstallReferrersFinished();
                     }
                 }
             });
-        }
-        catch(Exception e){
+        } catch(Exception e) {
             BranchLogger.e("Caught Exception SystemObserver fetchInstallReferrer " + e.getMessage());
-        }
-        finally {
-            if(callback != null){
+            if (callback != null) {
```

Comment:
While this should work, what was the thread order of 
`fetchLatestInstallReferrer`
`resumeWith`
and the `finally` block?

Reply:
I believe the order goes:
1. `fetchLatestInstallReferrer` is called as a coroutine
2. `resumeWith` is called when the coroutine finishes
3. `finally` usually gets called when `fetchLatestInstallReferrer` finishes, but i was seeing it get called right after the coroutine starts in this case.

Reply:
@gdeluna-branch After printing out the threads I can see the order is
1. `initTasks` on mqt_native_modules
2. `fetchInstallReferrer` on mqt_native_modules
3. `fetchInstallReferrer finally` block on mqt_native_modules
4. `fetchInstallReferrer` `resumeWith` on DefaultDispatcher-worker-1
5. `initTasks onInstallReferrersFinished` on DefaultDispatcher-worker-1